Packages/com.unity.inputsystem/InputSystem/Devices/InputDevice.cs

```diff
@@ -446,10 +446,53 @@ internal bool hasControlsWithDefaultState
         internal void SetUsage(InternedString usage)
         {
             // Make last entry in m_UsagesForEachControl be our device usage string.
+            SetUsage(m_UsageToControl != null ? m_UsageToControl.Length : 0, usage);
+        }
+
+        internal void AddUsage(InternedString usage)
+        {
+            // Check if the usage already exists
+            if (Array.Exists(m_UsagesForEachControl, o => o == usage))
```

Comment:
This also checks the usages for each control, not just for the device. Best way is probably to just iterate over `usages` on the device.

Also, the delegate likely allocates garbage. Would prefer to iterate directly here.